pkg/cloudprovider/providers/photon/photon.go

```diff
@@ -286,65 +301,122 @@ func (pc *PCCloud) List(filter string) ([]k8stypes.NodeName, error) {
 
 // NodeAddresses is an implementation of Instances.NodeAddresses.
```

Comment:
This comment doesn't really help me understand what the method does. Can you elaborate?